### PR TITLE
GIT_PREFIX_PATH - allow to specify subdirectory inside repository

### PR DESCRIPTION
**Pull request by krogon-dp**
This commit also encapsulate all variables in double quotes

### CODE REVIEW

**Comment by pieterlange** [comment]
Can you explain your usecase a bit more?

**Comment by krogon-dp** [comment]
1. Sharing single backup repository for different environments using subdirectories rather than branches.
2. Sharing repository that already may contain some data, which could be overwritten.

File: entrypoint.sh [hunk 1]
```diff
@@ -20,15 +22,16 @@ fi
 git config --global user.name "$GIT_USERNAME"
 git config --global user.email "$GIT_EMAIL"
 
-test -d /backup/git/ || git clone --depth 1 $GIT_REPO /backup/git --branch $GIT_BRANCH || git clone $GIT_REPO /backup/git
-cd /backup/git/
-git checkout ${GIT_BRANCH} || git checkout -b ${GIT_BRANCH}
+test -d "$GIT_REPO_PATH" || git clone --depth 1 "$GIT_REPO" "$GIT_REPO_PATH" --branch "$GIT_BRANCH" || git clone "$GIT_REPO" "$GIT_REPO_PATH"
+cd "$GIT_REPO_PATH"
+git checkout "${GIT_BRANCH}" || git checkout -b "${GIT_BRANCH}"
 git rm -r . || true
```
Comment (pieterlange) [inline entrypoint.sh:28]: Considering the intended purpose, you might want to be a little careful here :)

Comment (krogon-dp) [reply entrypoint.sh:28]: This part hasn't changed - only "/backup/git" was moved to a variable with name `GIT_REPO_PATH` to not hardcode anything.

Comment (pieterlange) [reply entrypoint.sh:28]: Look at me, reviewing PR's without my second coffee. ;)

**Comment by pieterlange** [comment]
Looks good and shouldn't break anything for other users. Thanks for the contribution!

**Pull request merged**